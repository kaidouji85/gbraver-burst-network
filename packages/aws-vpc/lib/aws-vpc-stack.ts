import { CfnOutput, Stack, StackProps } from "aws-cdk-lib";
import { aws_ec2 as ec2 } from "aws-cdk-lib";
import { Construct } from "constructs";

/** Cloud Formation Output プレフィックス */
const CFN_OUTPUT_PREFIX = 'VpcV2';

/** VPCスタックプロパティ */
interface VPCProps extends StackProps {
  /** サービス名 */
  service: string;
  /** VPC CIDR */
  cidr: string;
  /** AZ数 */
  maxAzs: number;
}

/** VPC スタック */
export class AwsVpcStack extends Stack {
  /**
   * @constructor
   * @param scope スコープ
   * @param id スタックのID
   * @param props CDKプロパティ
   */
  constructor(scope: Construct, id: string, props: VPCProps) {
    super(scope, id, props);
    const vpc = new ec2.Vpc(this, "vpc", {
      ipAddresses: ec2.IpAddresses.cidr(props.cidr),
      maxAzs: props.maxAzs,
      natGateways: 0,
    });
    new CfnOutput(this, "VpcId", {
      value: vpc.vpcId,
      exportName: `${props.service}:${CFN_OUTPUT_PREFIX}:VpcId`,
    });

    vpc.publicSubnets.forEach((subnets, index) => {
      new CfnOutput(this, `PublicNetAvailabilityZone${index}`, {
        value: subnets.availabilityZone,
        exportName: `${props.service}:${CFN_OUTPUT_PREFIX}:PublicNetAvailabilityZone${index}`,
      });
      new CfnOutput(this, `PublicSubnetId${index}`, {
        value: subnets.subnetId,
        exportName: `${props.service}:${CFN_OUTPUT_PREFIX}:PublicSubnetId${index}`,
      });
    });
  }
}
