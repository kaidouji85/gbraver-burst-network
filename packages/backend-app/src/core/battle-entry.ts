import type { ArmdozerId, PilotId } from "gbraver-burst-core";

import type { UserID } from "./user";
import type { WSAPIGatewayConnectionId } from "./ws-api-gateway-connection";

/**
 * ネット対戦エントリ
 * カジュアルマッチ、プライベートマッチ等、全ケースのエントリで
 * 必要最小限の情報をあつめた
 */
export type BattleEntry = {
  /** エントリーするユーザのID */
  userID: UserID;

  /** 選択したアームドーザID */
  armdozerId: ArmdozerId;

  /** 選択したパイロットID */
  pilotId: PilotId;

  /** コネクションID */
  connectionId: WSAPIGatewayConnectionId;
};
