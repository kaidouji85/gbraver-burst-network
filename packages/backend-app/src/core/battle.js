// @flow

import type { Player } from "gbraver-burst-core";
import type { GameState } from "gbraver-burst-core/lib/state/game-state";

import type { UserID } from "./user";
import type { WSAPIGatewayConnectionId } from "./ws-api-gateway-connection";

/** バトルID */
export type BattleID = string;

/** ステートが更新されるたびに発行されるユニークID */
export type FlowID = string;

/** バトルに参加するプレイヤー情報 */
export type BattlePlayer = Player & {
  /** ユーザID */
  userID: UserID,
  /** コネクションID */
  connectionId: WSAPIGatewayConnectionId,
};

/**
 * バトル情報
 * @template X プレイヤー情報
 */
export type Battle<X: BattlePlayer> = {
  /** バトルID */
  battleID: BattleID,
  /** フローID */
  flowID: FlowID,
  /** プレイヤー情報 */
  players: [X, X],
  /**
   * バトル更新ポーリングをするユーザのID
   * playersに含まれているユーザのIDを指定すること
   */
  poller: UserID,
  /** ステートヒストリー */
  stateHistory: GameState[],
};
