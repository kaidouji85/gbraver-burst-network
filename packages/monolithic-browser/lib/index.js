"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MonolithicBrowser = void 0;

var _login2 = require("./login");

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/** モノシリックサーバ ブラウザ用 SDK */
var MonolithicBrowser = /*#__PURE__*/function () {
  /**
   * コンストラクタ
   *
   * @param apiServerURL APIサーバURL
   */
  function MonolithicBrowser(apiServerURL) {
    _classCallCheck(this, MonolithicBrowser);

    this._apiServerURL = apiServerURL;
  }
  /**
   * ユーザID、パスワードでログインを行う
   * ログインに成功した場合はtrueを返す
   *
   * @param userID ユーザID
   * @param password パスワード
   * @return ログイン結果
   */


  _createClass(MonolithicBrowser, [{
    key: "login",
    value: function login(userID, password) {
      return (0, _login2.login)(userID, password, this._apiServerURL);
    }
  }]);

  return MonolithicBrowser;
}();

exports.MonolithicBrowser = MonolithicBrowser;