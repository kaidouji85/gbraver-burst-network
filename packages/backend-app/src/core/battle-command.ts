import { Command, CommandSchema } from "gbraver-burst-core";
import { z } from "zod";

import { BattleID, BattleIDSchema, FlowID, FlowIDSchema } from "./battle";
import { UserID, UserIDSchema } from "./user";

/** バトルコマンド */
export type BattleCommand = {
  /** ユーザID */
  userID: UserID;
  /** バトルID */
  battleID: BattleID;
  /** フローID */
  flowID: FlowID;
  /** コマンド */
  command: Command;
};

/** BattleCommand zodスキーマ */
export const BattleCommandSchema = z.object({
  userID: UserIDSchema,
  battleID: BattleIDSchema,
  flowID: FlowIDSchema,
  command: CommandSchema,
});