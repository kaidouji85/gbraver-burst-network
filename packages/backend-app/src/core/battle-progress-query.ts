import { BattleID, FlowID } from "./battle";

/** バトル進行 問合わせパラメータ */
export type BattleProgressQuery = {
  /** バトルID */
  battleID: BattleID;
  /** フローID */
  flowID: FlowID;
};