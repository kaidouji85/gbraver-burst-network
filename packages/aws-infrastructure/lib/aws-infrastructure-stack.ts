import * as cdk from '@aws-cdk/core';
import * as ec2 from "@aws-cdk/aws-ec2";
import * as ecs from "@aws-cdk/aws-ecs";
import * as ecr from "@aws-cdk/aws-ecr";
import * as iam from '@aws-cdk/aws-iam';

export class AwsInfrastructureStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const stage = new cdk.CfnParameter(this, 'stage', {
      type: 'String',
      description: 'serverless stage name',
    }).valueAsString;
    const websocketAPIID = cdk.Fn.importValue(`gbraver-burst-serverless:${stage}:WebsoketApiId`);
    const connectionsTableARN = cdk.Fn.importValue(`gbraver-burst-serverless:${stage}:ConnectionsTableArn`);
    const casualMatchEntriesTableARN = cdk.Fn.importValue(`gbraver-burst-serverless:${stage}:CasualMatchEntriesTableArn`);
    const battlesTableARN = cdk.Fn.importValue(`gbraver-burst-serverless:${stage}:BattlesTableArn`);

    const vpc = new ec2.Vpc(this, "vpc", {
      maxAzs: 1,
    });

    const matchMakeRepository = ecr.Repository.fromRepositoryName(this, "repo", "gbraver-burst-match-make");
    const matchMakePolicy = new iam.PolicyDocument({
      statements: [
        new iam.PolicyStatement({
          resources: [
            connectionsTableARN,
            casualMatchEntriesTableARN,
            battlesTableARN
          ],
          actions: [
            'dynamodb:PutItem',
            'dynamodb:GetItem',
            'dynamodb:DeleteItem',
            'dynamodb:Scan',
            'dynamodb:BatchWrite*',
          ],
        }),
        new iam.PolicyStatement({
          resources: ['arn:aws:execute-api:*:*:**/@connections/*'],
          actions: ['execute-api:ManageConnections'],
        }),
      ],
    });
    const matchMakeServiceTaskRole = new iam.Role(this, 'match-make-service-task-role', {
      roleName: 'ecs-service-task-role',
      assumedBy: new iam.ServicePrincipal('ecs-tasks.amazonaws.com'),
      inlinePolicies: {matchMakePolicy}
    });
    const matchMakeTaskDefinition = new ecs.TaskDefinition(this, "taskdef", {
      compatibility: ecs.Compatibility.FARGATE,
      cpu: "256",
      memoryMiB: "512",
      taskRole: matchMakeServiceTaskRole
    });
    const matchMakeLogging = new ecs.AwsLogDriver({
      streamPrefix: "gbraver-burst-match-make",
    });
    matchMakeTaskDefinition.addContainer(`match-make-container`, {
      image: ecs.ContainerImage.fromAsset('../serverless', {
        file: 'matchMake.Dockerfile'
      }),
      environment: {
        STAGE: stage,
        WEBSOCKET_API_ID: websocketAPIID,
      },
      logging: matchMakeLogging,
    });
    const cluster = new ecs.Cluster(this, "cluster", { vpc });
    const matchMakeService = new ecs.FargateService(this, "service", {
      cluster,
      taskDefinition: matchMakeTaskDefinition
    });
  }
}