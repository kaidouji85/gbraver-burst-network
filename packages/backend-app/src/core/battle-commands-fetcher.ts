import { BattlePlayer } from "./battle";
import { BattleCommand } from "./battle-command";

/** ゲーム参加プレイヤーのバトルコマンドを取得 */
export interface BattleCommandsFetcher {
  /**
   * ゲーム参加プレイヤーのバトルコマンドを取得する
   * @param players ゲーム参加プレイヤー
   * @return 取得結果、取得できなかった場合はnull
   */
  fetch(
    players: [BattlePlayer, BattlePlayer],
  ): Promise<[BattleCommand, BattleCommand] | null>;
}
