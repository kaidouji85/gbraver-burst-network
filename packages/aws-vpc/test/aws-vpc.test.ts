import { expect as expectCDK, matchTemplate, MatchStyle } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import * as AwsVpc from '../lib/aws-vpc-stack';

// TODO テストに失敗しないようにexpectCDKを正しく定義する
test('Empty Stack', () => {
    const app = new cdk.App();
    // WHEN
    const stack = new AwsVpc.AwsVpcStack(app, 'MyTestStack', {
      service: 'my-service'
    });
    // THEN
    expectCDK(stack).to(matchTemplate({
      "Resources": {}
    }, MatchStyle.EXACT))
});
