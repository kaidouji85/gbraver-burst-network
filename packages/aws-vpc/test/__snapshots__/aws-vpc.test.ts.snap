// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VPCスタックがスナップショットと一致している 1`] = `
{
  "Outputs": {
    "PublicNetAvailabilityZone0": {
      "Export": {
        "Name": "gbraver-buesr-sls-dev-gXXXXX:PublicNetAvailabilityZone0",
      },
      "Value": {
        "Fn::Select": [
          0,
          {
            "Fn::GetAZs": "",
          },
        ],
      },
    },
    "PublicNetAvailabilityZone1": {
      "Export": {
        "Name": "gbraver-buesr-sls-dev-gXXXXX:PublicNetAvailabilityZone1",
      },
      "Value": {
        "Fn::Select": [
          1,
          {
            "Fn::GetAZs": "",
          },
        ],
      },
    },
    "PublicSubnetId0": {
      "Export": {
        "Name": "gbraver-buesr-sls-dev-gXXXXX:PublicSubnetId0",
      },
      "Value": {
        "Ref": "vpcPublicSubnetSubnet1Subnet031C8F62",
      },
    },
    "PublicSubnetId1": {
      "Export": {
        "Name": "gbraver-buesr-sls-dev-gXXXXX:PublicSubnetId1",
      },
      "Value": {
        "Ref": "vpcPublicSubnetSubnet2Subnet591FB8AD",
      },
    },
    "SubnetCount": {
      "Export": {
        "Name": "gbraver-buesr-sls-dev-gXXXXX:SubnetCount",
      },
      "Value": "2",
    },
    "VpcId": {
      "Export": {
        "Name": "gbraver-buesr-sls-dev-gXXXXX:VpcId",
      },
      "Value": {
        "Ref": "vpcA2121C38",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "vpcA2121C38": {
      "Properties": {
        "CidrBlock": "172.16.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": [
          {
            "Key": "Name",
            "Value": "gbraver-buesr-sls-dev-gXXXXX/vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "vpcEIGW693EAFBD1": {
      "Properties": {
        "VpcId": {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::EgressOnlyInternetGateway",
    },
    "vpcIGWE57CBDCA": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "gbraver-buesr-sls-dev-gXXXXX/vpc",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "vpcPublicSubnetSubnet1DefaultRoute659C75CAD": {
      "DependsOn": [
        "vpcipv6cidr41CEB63E",
      ],
      "Properties": {
        "DestinationIpv6CidrBlock": "::/0",
        "GatewayId": {
          "Ref": "vpcIGWE57CBDCA",
        },
        "RouteTableId": {
          "Ref": "vpcPublicSubnetSubnet1RouteTable8370FADE",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPublicSubnetSubnet1DefaultRoute88AEC4F6": {
      "DependsOn": [
        "vpcipv6cidr41CEB63E",
        "vpcVPCGW7984C166",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "vpcIGWE57CBDCA",
        },
        "RouteTableId": {
          "Ref": "vpcPublicSubnetSubnet1RouteTable8370FADE",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPublicSubnetSubnet1RouteTable8370FADE": {
      "DependsOn": [
        "vpcipv6cidr41CEB63E",
      ],
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "gbraver-buesr-sls-dev-gXXXXX/vpc/PublicSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcPublicSubnetSubnet1RouteTableAssociation03099B2F": {
      "DependsOn": [
        "vpcipv6cidr41CEB63E",
      ],
      "Properties": {
        "RouteTableId": {
          "Ref": "vpcPublicSubnetSubnet1RouteTable8370FADE",
        },
        "SubnetId": {
          "Ref": "vpcPublicSubnetSubnet1Subnet031C8F62",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcPublicSubnetSubnet1Subnet031C8F62": {
      "DependsOn": [
        "vpcipv6cidr41CEB63E",
      ],
      "Properties": {
        "AssignIpv6AddressOnCreation": true,
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "172.16.0.0/24",
        "Ipv6CidrBlock": {
          "Fn::Select": [
            0,
            {
              "Fn::Cidr": [
                {
                  "Fn::Select": [
                    0,
                    {
                      "Fn::GetAtt": [
                        "vpcA2121C38",
                        "Ipv6CidrBlocks",
                      ],
                    },
                  ],
                },
                2,
                "64",
              ],
            },
          ],
        },
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "PublicSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          {
            "Key": "Name",
            "Value": "gbraver-buesr-sls-dev-gXXXXX/vpc/PublicSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcPublicSubnetSubnet2DefaultRoute24DE7C87": {
      "DependsOn": [
        "vpcipv6cidr41CEB63E",
        "vpcVPCGW7984C166",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "vpcIGWE57CBDCA",
        },
        "RouteTableId": {
          "Ref": "vpcPublicSubnetSubnet2RouteTable187EC9B4",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPublicSubnetSubnet2DefaultRoute6B868AF84": {
      "DependsOn": [
        "vpcipv6cidr41CEB63E",
      ],
      "Properties": {
        "DestinationIpv6CidrBlock": "::/0",
        "GatewayId": {
          "Ref": "vpcIGWE57CBDCA",
        },
        "RouteTableId": {
          "Ref": "vpcPublicSubnetSubnet2RouteTable187EC9B4",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPublicSubnetSubnet2RouteTable187EC9B4": {
      "DependsOn": [
        "vpcipv6cidr41CEB63E",
      ],
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "gbraver-buesr-sls-dev-gXXXXX/vpc/PublicSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcPublicSubnetSubnet2RouteTableAssociation86760EC0": {
      "DependsOn": [
        "vpcipv6cidr41CEB63E",
      ],
      "Properties": {
        "RouteTableId": {
          "Ref": "vpcPublicSubnetSubnet2RouteTable187EC9B4",
        },
        "SubnetId": {
          "Ref": "vpcPublicSubnetSubnet2Subnet591FB8AD",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcPublicSubnetSubnet2Subnet591FB8AD": {
      "DependsOn": [
        "vpcipv6cidr41CEB63E",
      ],
      "Properties": {
        "AssignIpv6AddressOnCreation": true,
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "172.16.1.0/24",
        "Ipv6CidrBlock": {
          "Fn::Select": [
            1,
            {
              "Fn::Cidr": [
                {
                  "Fn::Select": [
                    0,
                    {
                      "Fn::GetAtt": [
                        "vpcA2121C38",
                        "Ipv6CidrBlocks",
                      ],
                    },
                  ],
                },
                2,
                "64",
              ],
            },
          ],
        },
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "PublicSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          {
            "Key": "Name",
            "Value": "gbraver-buesr-sls-dev-gXXXXX/vpc/PublicSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcVPCGW7984C166": {
      "Properties": {
        "InternetGatewayId": {
          "Ref": "vpcIGWE57CBDCA",
        },
        "VpcId": {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
    "vpcipv6cidr41CEB63E": {
      "Properties": {
        "AmazonProvidedIpv6CidrBlock": true,
        "VpcId": {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::VPCCidrBlock",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
